scraper/main.py

```diff
@@ -1,2 +1,61 @@
-if __name__ == "__main__":
-    print("Hello World")
+import requests
+from bs4 import BeautifulSoup
+
+# URLs for FAANG internship listings
+urls = {
+    'Facebook': 'https://www.facebook.com/careers/jobs?search_text=internship',
+    'Amazon': 'https://www.amazon.jobs/en/search?base_query=internship',
+    'Apple': 'https://jobs.apple.com/en-ca/search?team=internships-STDNT-INTRN',
+    'Netflix': 'https://jobs.netflix.com/search?q=internship',
+    'Google': 'https://careers.google.com/jobs/results/?q=internship'
+}
+
+def get_facebook_internships(soup):
+    internship_titles = [container.text for container in soup.select('._af0h a:first-child div:first-child div:first-child div:first-child div:first-child')]
+    internship_locations = [container.get('data-tooltip-content') for container in soup.select('._8seh div:first-child div:first-child div:first-child')]
+    internship_links = ['https://www.metacareers.com' + container.get('href') for container in soup.select('._af0h a:first-child')]
+    internships = [(title, location.split('\n'), link) for title in internship_titles for location in internship_locations for link in internship_links]
+    return internships
+
+def get_amazon_internships(soup):
+    internship_titles = [container.text for container in soup.select('.row div:first-child')]
+    internships = []
+    return internships
+
+def get_apple_internships(soup):
+    internship_titles = [i.text for i in soup.select('.table--advanced-search__title')]
+    internship_locations = [i.text for i in soup.select('.table-col-2 span:first-child')]
+    internship_links = ['https://jobs.apple.com' + container.get('href') for container in soup.select('.table--advanced-search__title')]
+    internships = [(title, location, link) for title in internship_titles for location in internship_locations for link in internship_links]
+    return internships
```

Comment:
I ran the code and Apple is the only site giving output however there is a lot more output then should be expected. I think this is due to the way you are creating the returned internships array.
You are using list comprehension to create a list of tuples `[ (title, location, link) ]` but to create each tuple you are iterating through each sub-list like this:
```python
for title in internship_titles:
  for location in internship_locations:
    for link in internship_links:
```
so a tuple is being created for each iteration of each loop causing duplicate data.
i.e.
```python
titles = [ title1, title2, title3 ]
locations = [ location1, location2, location3]
links = [ link1, link2, link3]

# will return 
[ (title1, location1, link1),
(title1, location1, link2),
(title1, location1, link3),
(title1, location2, link1)
.
.
(title3,location3,link3)]
```
I would recommend taking a look at creating a dictionary for each job container that way you only have to iterate through a single job post once and you assign all necessary information in key, value pairs. You can then create an array of these dictionaries and iterating now becomes linear O(n) as opposed to O(n<sup>3</sup>)

Reply:
Sounds good, I'll look into it.

Reply:
@a-camarillo I changed the comprehension slightly since it wasn't returning the correct output to the following:

`
internships = [(internship_titles[i], internship_locations[i], internship_links[i]) for i in range(len(internship_titles))]
`

This should be O(n). I can change it to a dictionary but I'm not sure why a dictionary would be better. Specifically, if the keys are the job titles and the user wants to filter by location or some other attribute, I don't see any pros in using a dictionary. 

The comprehension above doesn't work for Facebook but that's just because the length of the lists `internship_titles`, `internship_locations`, and `internship_links` are not the same, which they shouldn't be.

Reply:
For sure, just a suggestion on a different approach. 

As far as filtering, you won't have to worry about that for the scraper since the data is going to be retrieved from the database. What the scraper should accomplish is retrieving the necessary information and inserting the information into the database. How that information actually gets presented to a user will be handled between the UI and the database.